packages/flutter/test/material/snack_bar_theme_test.dart

```diff
@@ -186,6 +186,114 @@ void main() {
     expect(button.text.style!.color, textColor);
   });
 
+  testWidgets('SnackBarAction uses actionBackgroundData', (WidgetTester tester) async {
+    final MaterialStateColor actionBackgroundColor = MaterialStateColor.resolveWith((Set<MaterialState> states) {
+      if (states.contains(MaterialState.disabled)) {
+        return Colors.blue;
+      }
+      return Colors.purple;
+    });
+
+    await tester.pumpWidget(MaterialApp(
+      theme: ThemeData(snackBarTheme: _createSnackBarTheme(actionBackgroundColor: actionBackgroundColor)),
+      home: Scaffold(
+        body: Builder(
+          builder: (BuildContext context) {
+            return GestureDetector(
+              onTap: () {
+                ScaffoldMessenger.of(context).showSnackBar(SnackBar(
+                  content: const Text('I am a snack bar.'),
+                  action: SnackBarAction(
+                    label: 'ACTION',
+                    onPressed: () {},
+                  ),
+                ));
+              },
+              child: const Text('X'),
+            );
+          },
+        ),
+      ),
+    ));
+
+    await tester.tap(find.text('X'));
+    await tester.pump(); // start animation
+    await tester.pump(const Duration(milliseconds: 750));
```

Comment:
Does a `tester.pumpAndSettle` work instead of a hardcoded duration? If not, could you add a comment describing why we're using the hardcoded duration here instead of `pumpAndSettle`?

Reply:
I saw that you caught a different instance of a hardcoded duration and fixed it, good catch! It looks like this instance didn't get fixed though.